hightime/_datetime.py

```diff
@@ -0,0 +1,427 @@
+            kwargs = dict()
+            if _PY36:
+                kwargs["timespec"] = timespec
+            return super().isoformat(sep, **kwargs)
+
+        value = self._yoctosecond + (self._femtosecond * 1000000000)
+        for spec in specs:
+            if spec == timespec:
+                break
+            value //= 1000
+
+        fmt = specs[timespec]
+        iso_strs = super().isoformat(sep).split("+")
+        iso_strs[0] = iso_strs[0].split(".")[0]
+        iso_strs[0] += "." + fmt.format(self.microsecond, value)
+        return "+".join(iso_strs)
+
+    if not _PY36:
+        _isoformat = isoformat
+
+        def isoformat(self, sep="T"):
+            return self._isoformat(sep)
+
+    def replace(
+        self,
+        year=None,
+        month=None,
+        day=None,
+        hour=None,
+        minute=None,
+        second=None,
+        microsecond=None,
+        femtosecond=None,
+        yoctosecond=None,
+        tzinfo=True,
+        **kwargs
+    ):
+        if year is None:
+            year = self.year
+        if month is None:
+            month = self.month
+        if day is None:
+            day = self.day
+        if hour is None:
+            hour = self.hour
+        if minute is None:
+            minute = self.minute
+        if second is None:
+            second = self.second
+        if microsecond is None:
+            microsecond = self.microsecond
+        if femtosecond is None:
+            femtosecond = self.femtosecond
+        if yoctosecond is None:
+            yoctosecond = self.yoctosecond
+        if tzinfo is True:
+            tzinfo = self.tzinfo
+        if _PY36:
+            kwargs.setdefault("fold", self.fold)
+        return type(self)(
+            year,
+            month,
+            day,
+            hour,
+            minute,
+            second,
+            microsecond,
+            femtosecond,
+            yoctosecond,
+            tzinfo=tzinfo,
+            **kwargs
+        )
+
+    # String operators
+
+    def __repr__(self):
+        """Convert to formal string, for repr()."""
+        s = "{}.{}".format(self.__class__.__module__, self.__class__.__qualname__)
+
+        # We only expose subminute fields if they aren't 0
+        subsecond_fields = reversed(
+            list(
+                dropwhile(
+                    lambda x: x == 0,
+                    [
+                        self.yoctosecond,
+                        self.femtosecond,
+                        self.microsecond,
+                        self.second,
+                    ],
+                )
+            )
+        )
+
+        values = [self.year, self.month, self.day, self.hour, self.minute] + list(
+            subsecond_fields
+        )
+        s += "({}".format(", ".join(map(str, values)))
+
+        if self.tzinfo is not None:
+            s += ", tzinfo={!r}".format(self.tzinfo)
+        if getattr(self, "fold", 0):
+            s += ", fold=1"
+        s += ")"
```

Comment:
In the pure-python implementation `tzinfo` comes first: https://github.com/python/cpython/blob/master/Lib/datetime.py#L1399

In the C implementation `tzinfo` comes last:
https://github.com/python/cpython/blob/master/Modules/_datetimemodule.c#L5367

🤷‍♂️

Reply:
Interesting. It doesn't matter either way since they are all keyword args. Given pure-python implementation is the newer one, makes sense to follow that.